src/wp-includes/capabilities.php

```diff
@@ -801,6 +801,30 @@ function map_meta_cap( $cap, $user_id, ...$args ) {
 		case 'delete_app_password':
 			$caps = map_meta_cap( 'edit_user', $user_id, $args[0] );
 			break;
+		case 'edit_block_binding':
+			/*
+			 * If the post ID is null, check if the context is the site editor.
+			 * Fall back to the edit_theme_options in that case.
+			 */
+			if ( ! isset( $args[0] ) ) {
+				$screen = get_current_screen();
+				if ( ! isset( $screen->id ) || 'site-editor' !== $screen->id ) {
```

Comment:
If the post id gets passed as an argument, then probably it's easier to pass also the name of the editor:

https://github.com/WordPress/wordpress-develop/blob/7f4fd30dffdfe8d7d817c2cdb4ab98408f1c0add/src/wp-includes/class-wp-block-editor-context.php#L30

Or even the entire block editor context.

Reply:
I've used the entire block editor context as suggested in this commit: https://github.com/WordPress/wordpress-develop/pull/7258/commits/a6a8d21393118d53ce91c636c9676613042fc9b8